client/my-sites/posts/main.jsx

```diff
@@ -1,46 +1,97 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React from 'react';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
-var PostsNavigation = require( './posts-navigation' ),
-	observe = require( 'lib/mixins/data-observe' ),
-	SidebarNavigation = require( 'my-sites/sidebar-navigation' ),
-	PostList = require( './post-list' ),
-	config = require( 'config' ),
-	Main = require( 'components/main' ),
-	notices = require( 'notices' );
+import PostsNavigation from './posts-navigation';
+import observe from 'lib/mixins/data-observe';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+import PostList from './post-list';
+import config from 'config';
+import Main from 'components/main';
+import notices from 'notices';
+import QueryPosts from 'components/data/query-posts';
+import QueryPostCounts from 'components/data/query-post-counts';
+import Draft from 'my-sites/draft';
+import { getSelectedSite } from 'state/ui/selectors';
+import {
+	getSitePostsForQueryIgnoringPage,
+	isRequestingSitePostsForQuery
+} from 'state/posts/selectors';
+import Button from 'components/button';
+import Card from 'components/card';
+import Count from 'components/count';
+import Gridicon from 'components/gridicon';
+import NoResults from 'my-sites/no-results';
+import { sectionify } from 'lib/route/path';
+import { getAllPostCount } from 'state/posts/counts/selectors';
 
-module.exports = React.createClass( {
+const PostsMain = React.createClass( {
+	mixins: [ observe( 'sites' ) ],
 
-	displayName: 'Posts',
+	componentWillMount() {
+		const selectedSite = this.props.sites.getSelectedSite();
+		this.setWarning( selectedSite );
+	},
 
-	mixins: [ observe( 'sites' ) ],
+	componentWillReceiveProps( nextProps ) {
+		const selectedSite = nextProps.sites.getSelectedSite();
+		this.setWarning( selectedSite );
+	},
+
+	mostRecentDrafts() {
+		const site = this.props.sites.getSelectedSite();
+		const isLoading = this.props.draftCount !== 0 && this.props.loadingDrafts;
 
-	componentWillMount: function() {
-		var selectedSite = this.props.sites.getSelectedSite();
-		this._setWarning( selectedSite );
+		return (
+			<div className="posts__recent-drafts">
+				<QueryPosts
+					siteId={ site && site.ID }
+					query={ this.props.draftsQuery } />
+				<QueryPostCounts siteId={ site && site.ID } type="post" />
+				<Card compact className="posts__drafts-header">
+					<span>
+						{ this.translate( 'Drafts' ) }
+						{ this.props.draftCount ? <Count count={ this.props.draftCount } /> : null }
+					</span>
+					<Button disabled={ ! site } borderless href={ `/post/${ site.slug }` }>
+						<Gridicon icon="plus" />
+					</Button>
+				</Card>
+				{ this.props.drafts && this.props.drafts.map( this.renderDraft, this ) }
```

Comment:
Do you need the second argument to `map`? Forget if this is included in `React.createClass` auto-binding.